mlir-sycl/lib/Dialect/SYCL/Utils/Utils.cpp

```diff
@@ -49,3 +58,21 @@ sycl::createSYCLAccessorSubscriptOp(AccessorPtrValue accessor,
       builder.getI64IntegerAttr(targetToAddressSpace(accTy.getTargetMode())));
   return builder.create<SYCLAccessorSubscriptOp>(loc, MT, accessor, id);
 }
+
+void sycl::populateWorkGroupSize(SmallVectorImpl<Value> &wgSizes,
+                                 unsigned numDims, OpBuilder builder) {
+  const auto arrayType = builder.getType<sycl::ArrayType>(
+      numDims, MemRefType::get(numDims, builder.getIndexType()));
+  const auto rangeTy = builder.getType<sycl::RangeType>(numDims, arrayType);
+  Location loc = builder.getUnknownLoc();
+  auto wgSize = builder.create<sycl::SYCLWorkGroupSizeOp>(loc, rangeTy);
```

Comment:
As I mentioned in my previous code review please create a function here called `createWorkGroupSize` and move back this function in LoopInternalization.

Reply:
I thought you want `populateWorkGroupSize` to be moved to Utils.

Reply:
Added.

Reply:
My opinion is that `populateWorkGroupSize` is a specialized function that is useful to the LoopInternalization pass. Not a generic utility function that would be useful to the SYCL dialect. 

Perhaps other ppl will have a different opinion, but that is how I feel.

Reply:
No it is up to the person that has a question to resolve a thread. Not to the author of the PR.

Reply:
ok, let's keep the two threads then.